### PR TITLE
Add category integration

### PR DESCRIPTION
**Pull request by YaroslavYarmoshyk**

### CODE REVIEW

File: src/main/java/com/online/bookstore/controller/CategoryResource.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    @GetMapping
+    @Operation(summary = "Get all categories", description = "Get all categories")
+    public List<CategoryDto> getAllCategories() {
+        return categoryService.getAllCategories();
+    }
+
+    @GetMapping(value = "/{id}")
+    @Operation(summary = "Get category by ID", description = "Get specific category")
+    public CategoryDto getAllCategories(final @PathVariable Long id) {
```
Comment (Vasul-Pankiv) [inline src/main/java/com/online/bookstore/controller/CategoryResource.java]: naming)

File: src/main/java/com/online/bookstore/repository/BookRepository.java [hunk 2]
```diff
@@ -1,7 +1,27 @@
...
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
+
```
Comment (Vasul-Pankiv) [inline src/main/java/com/online/bookstore/repository/BookRepository.java]: do not start class/interface with empty line (fix everywhere)

File: src/main/resources/db/migration/V6__add_categories.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
+CREATE TABLE IF NOT EXISTS categories
```
Comment (Vasul-Pankiv) [inline src/main/resources/db/migration/V6__add_categories.sql:1]: better to create diff migration for creating tables and inserting data there

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/main/java/com/online/bookstore/controller/CategoryResource.java (hunk 1 above)
Comment (YaroslavYarmoshyk) [reply src/main/java/com/online/bookstore/controller/CategoryResource.java]: fixed

File: src/main/java/com/online/bookstore/repository/BookRepository.java (hunk 2 above)
Comment (YaroslavYarmoshyk) [reply src/main/java/com/online/bookstore/repository/BookRepository.java]: rewritten

File: src/main/resources/db/migration/V6__add_categories.sql (hunk 3 above)
Comment (YaroslavYarmoshyk) [reply src/main/resources/db/migration/V6__add_categories.sql:1]: sure, thanks

**Pull request merged**